### ISSUE TITLE
Psalm fails to detect issue with trait method having few params

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#803]
This should emit an error, but doesn’t

```php
trait T {
  protected function foo() : void {}
  
  public function bat() : void {
      $this->foo();
  }
}

class C {
  use T;
  
  protected function foo(string $s) : void {}
}
```

This should remain fine:

```php
trait T {
  protected function foo() : void {}
  
  public function bat() : void {
      $this->foo();
  }
}

class C {
  use T;
  
  protected function foo(string $s) : void {}
  
  public function bat() : void {
      $this->foo("bat");
  }
}
```
